packages/babel-parser/src/util/class-scope.js

```diff
@@ -19,21 +19,25 @@ export class ClassScope {
   undefinedPrivateNames: Map<string, number> = new Map();
 }
 
-export default class ClassScopeHandler {
-  stack: Array<ClassScope> = [];
+export default class ClassScopeHandler<IClassScope: ClassScope = ClassScope> {
+  stack: Array<IClassScope> = [];
   declare raise: raiseFunction;
   undefinedPrivateNames: Map<string, number> = new Map();
 
   constructor(raise: raiseFunction) {
     this.raise = raise;
   }
 
-  current(): ClassScope {
+  current(): IClassScope {
     return this.stack[this.stack.length - 1];
   }
 
   enter() {
-    this.stack.push(new ClassScope());
+    this.stack.push(this.createScope());
```

Comment:
`Cannot call 'this.stack.push' because 'ClassScope' [1] is incompatible with 'IClassScope' [2] in array element.`
Why this?